app/src/test/kotlin/com/wire/android/feature/conversation/usecase/ResetCurrentConversationIdUseCaseTest.kt

```diff
@@ -0,0 +1,53 @@
+package com.wire.android.feature.conversation.usecase
+
+import com.wire.android.UnitTest
+import com.wire.android.core.exception.EmptyCacheFailure
+import com.wire.android.core.functional.Either
+import com.wire.android.feature.conversation.data.ConversationRepository
+import io.mockk.coEvery
+import io.mockk.coVerify
+import io.mockk.impl.annotations.MockK
+import kotlinx.coroutines.runBlocking
+import org.junit.Before
+import org.junit.Test
+import org.mockito.ArgumentMatchers.any
+
+class ResetCurrentConversationIdUseCaseTest : UnitTest() {
+
+    private lateinit var resetCurrentConversationIdUseCase: ResetCurrentConversationIdUseCase
+
+    @MockK
+    private lateinit var conversationRepository: ConversationRepository
+
+    @Before
+    fun setUp() {
+        resetCurrentConversationIdUseCase = ResetCurrentConversationIdUseCase(conversationRepository)
+    }
+
+    @Test
+    fun `given the repository succeeds when resetting the current conversation id, then the success is propagated`() = runBlocking {
+        coEvery { conversationRepository.restCurrentConversationId() } returns Either.Right(Unit)
+
+        val result = resetCurrentConversationIdUseCase.run(Unit)
+        assert(true) {result == Either.Right(Unit)}
+
+        coVerify(exactly = 1) { conversationRepository.restCurrentConversationId() }
+    }
+
+    @Test
+    fun `given the repository fails, when resetting the current conversation id, then the failure is propagate`() = runBlocking {
+        coEvery { conversationRepository.restCurrentConversationId() } returns Either.Left(EmptyCacheFailure)
+
+        val result = resetCurrentConversationIdUseCase.run(Unit)
+        assert(true) {result == Either.Left(EmptyCacheFailure)}
+
```

Comment:
On top of the extension for assertion, I suggest mocking the failure. If `EmptyCacheFailure` implementation changes and it becomes something other than an object, it would break these tests. Also, you guarantee that whatever the repository is throwing is being propagated exactly as it is.

```suggestion
        val failure = mockk<Failure>()
        coEvery { conversationRepository.restCurrentConversationId() } returns Either.Left(failure)

        val result = resetCurrentConversationIdUseCase.run(Unit)

        result.shouldFail {
            it shouldBeEqualTo failure
        }
```